src/main/java/org/openpnp/machine/reference/feeder/ReferenceTrayFeeder.java

```diff
@@ -71,27 +72,32 @@ public void feed(Nozzle nozzle)
 			throws Exception {
 		logger.debug("{}.feed({})", getName(), nozzle);
 		int partX, partY;
+		double JavaBugWorkaroundX,JavaBugWorkaroundY,JavaBugWorkaroundC;
 		
 		if (feedCount >= (trayCountX * trayCountY)) {
 		    throw new Exception(String.format("Tray empty on feeder %s.", getName()));
 		}
         
+	JavaBugWorkaroundC=feedCount; JavaBugWorkaroundY=trayCountY; JavaBugWorkaroundX=trayCountX;
         if (trayCountX >= trayCountY) {
             // X major axis.
-            partX = feedCount / trayCountY;
-            partY = feedCount % trayCountY;
+            JavaBugWorkaroundX = JavaBugWorkaroundC / JavaBugWorkaroundY;
+            JavaBugWorkaroundY = JavaBugWorkaroundC % JavaBugWorkaroundY;
         }
         else {
             // Y major axis.
-            partX = feedCount % trayCountX;
-            partY = feedCount / trayCountX;
+            JavaBugWorkaroundY = JavaBugWorkaroundC / JavaBugWorkaroundX;
+            JavaBugWorkaroundX = JavaBugWorkaroundC % JavaBugWorkaroundX;
         }
+	partX=(int)JavaBugWorkaroundX;
+	partY=(int)JavaBugWorkaroundY;
         
         // Multiply the offsets by the X/Y part indexes to get the total offsets
         // and then add the pickLocation to offset the final value.
         // and then add them to the location to get the final pickLocation.
+        
         pickLocation = location.add(
-                offsets.multiply(partX, partY, 0.0, 0.0));
+                offsets.multiply(partX, partY, 1.0, 1.0));
```

Comment:
I have traced it down, problem was upgrade to java8 have not
overridded all env varaibles
and so java7 core and java8 was mixed and resulted in integer modulus bug.
moving it on floating point domain resolved this issue apparently.

For the behaviour change.
It's for part stacks, and no, it don't break anything, as offsets is
always initialized with 0
and the wizard don't allow to change C and Z.
It is possible to change it on machine.xml without gui.
If you stacked parts, like metal dome caps for flexible keyboard, or
sim/sd cards, ...
with this modification is it possible to have one pick position, that
load parts having different
Z with each feed and thus having one part on top of the other.
Sometime parts need stacked with 180 degree offset because of pin
heigh differences of one side, that otherwise accumulated too mutch.
It makes only sense with one single pick position. If you have as
example two stacks, it's
not possible to configure this as single feeder, two separate feeder
must be configured for the same part.

2016-01-23 0:41 GMT+01:00, Arvid E. Picciani <notifications@github.com>:
>>          pickLocation = location.add(
>> -                offsets.multiply(partX, partY, 0.0, 0.0));
>> +                offsets.multiply(partX, partY, 1.0, 1.0));
>
> That's a behaviour change here. Could you explain what its for, so we can
> check it doesnt break anything?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openpnp/openpnp/pull/195/files#r50606149